climakitae/threshold_tools.py

```diff
@@ -0,0 +1,613 @@
+                lowest_aicc_distr = aicc_gum[0]
+            elif lowest_aicc_value == aicc_wei[1]:
+                lowest_aicc_distr = aicc_wei[0]
+            elif lowest_aicc_value == aicc_pe3[1]:
+                lowest_aicc_distr = aicc_pe3[0]
+            elif lowest_aicc_value == aicc_gpa[1]:
+                lowest_aicc_distr = aicc_gpa[0]
+
+            return all_aicc_results, lowest_aicc_distr, lowest_aicc_value
+
+        return xr.apply_ufunc(
+            function,
+            ams,
+            input_core_dims=[["time"]],
+            exclude_dims=set(("time",)),
+            output_core_dims=[[], [], []],
+        )
+
+    all_aicc_results, lowest_aicc_distr, lowest_aicc_value = apply_function(ams)
+
+    all_aicc_results = all_aicc_results.rename("all_aicc_results")
+    new_ds = all_aicc_results.to_dataset()
+    new_ds["lowest_aicc_distr"] = lowest_aicc_distr
+    new_ds["lowest_aicc_value"] = lowest_aicc_value
+
+    if multiple_points == True:
+        new_ds = new_ds.unstack("allpoints")
+
+    new_ds.attrs = ams_attributes
+    new_ds.attrs["model selection technique"] = "AICc"
+
+    return new_ds
+
+
+#####################################################################
+
+
+def get_ks_stat(ams, distr="gev", multiple_points=True):
+
+    distrs = ["gev", "gumbel", "weibull", "pearson3", "genpareto"]
+    if distr not in distrs:
+        raise ValueError(
+            "invalid distr type. expected one of the following: %s" % distrs
+        )
+
+    ams_attributes = ams.attrs
+
+    if multiple_points == True:
+        ams = ams.stack(allpoints=["x", "y"]).squeeze().groupby("allpoints")
+
+    def apply_function(ams):
+        def function(ams):
+
+            if distr == "gev":
+                try:
+                    lmoments = ldistr.gev.lmom_fit(ams)
+                    fitted_distr = stats.genextreme(**lmoments)
+                    ks = stats.kstest(
+                        ams,
+                        "genextreme",
+                        args=(lmoments["c"], lmoments["loc"], lmoments["scale"]),
+                    )
+                    d_statistic = ks[0]
+                    p_value = ks[1]
+                except ValueError:
+                    d_statistic = np.nan
+                    p_value = np.nan
+
+            if distr == "gumbel":
+                try:
+                    lmoments = ldistr.gum.lmom_fit(ams)
+                    fitted_distr = stats.gumbel_r(**lmoments)
+                    ks = stats.kstest(
+                        ams, "gumbel_r", args=(lmoments["loc"], lmoments["scale"])
+                    )
+                    d_statistic = ks[0]
+                    p_value = ks[1]
+                except ValueError:
+                    d_statistic = np.nan
+                    p_value = np.nan
+
+            if distr == "weibull":
+                try:
+                    lmoments = ldistr.wei.lmom_fit(ams)
+                    fitted_distr = stats.weibull_min(**lmoments)
```

Comment:
Just a note that the weibull_min is most common, and IMO the best choice

Reply:
that makes sense! for extreme value analysis, at least, it seems like the literature was pointing towards the GEV as being the default so went with that. But if you feel good about weibull_min being the one that's most commonly used by utilities, i'm happy to set this as the default. let me know!